### PR TITLE
Clamp Shape node start/end to [0, rank] per ONNX spec

### LINKED ISSUE #216: Shape node: missing clamping of start/end to [0, rank] per spec
## Summary

The ONNX spec for the Shape operator explicitly states that `start` and `end` values should be clamped to `[0, rank]`, but the implementation only normalizes negative indices. Out-of-bounds positive values pass through unchecked, which could lead to panics or incorrect results.

## Location

`crates/onnx-ir/src/node/shape.rs`, line 127

## Suggested fix

After normalizing negative indices, add clamping:

```rust
start_dim = start_dim.max(0).min(rank as i64);
end_dim = end_dim.max(0).min(rank as i64);
```

Add tests for `start > rank` and `end > rank` scenarios.

### PR DESCRIPTION
**Pull request by antimora**
## Summary
- Add clamping of `start` and `end` to `[0, rank]` after negative index normalization in the Shape node processor, as required by the ONNX spec
- Remove stale FIXME/TODO comments
- Add 4 unit tests covering out-of-bounds edge cases (start > rank, end > rank, both > rank, negative values below -rank)

Fixes #216

## Test plan
- [x] `cargo test -p onnx-ir -- shape` passes (20 tests including 4 new ones)
- [ ] Verify no regressions in `cargo test -p onnx-tests`

### CODE REVIEW

File: crates/onnx-ir/src/node/shape.rs [hunk 1]
```diff
@@ -132,10 +128,10 @@ impl NodeProcessor for ShapeProcessor {
...
         }
 
-        // TODO: Missing clamping to [0, rank] as per ONNX spec - out-of-bounds positive values are not clamped.
-        // The spec explicitly states: "Values are clamped to [0, rank]" but implementation only normalizes negative indices.
-        // This could lead to panics or incorrect results when start/end exceed rank.
-        // Should add: start_dim = start_dim.max(0).min(rank); end_dim = end_dim.max(0).min(rank);
+        // Clamp to [0, rank] per ONNX spec
+        let rank_i64 = rank as i64;
+        start_dim = start_dim.max(0).min(rank_i64);
+        end_dim = end_dim.max(0).min(rank_i64);
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/shape.rs]: Valid observation, but this is pre-existing behavior unrelated to the clamping fix. In practice, ONNX models with opset < 15 will not have start/end attributes, so the defaults (start=0, end=rank) apply and the behavior is correct. Opset-aware attribute gating could be a separate issue if desired.

I did address the suppressed comment about start > end underflow: end_dim is now clamped to max(start_dim) so the subtraction cannot underflow (dcde3f0e).

[the remark this answers is not in the document]
Comment (antimora) [reply crates/onnx-ir/src/node/shape.rs]: Fixed in d9ab491a. `extract_config` now gates attribute extraction behind `if opset >= 15`, so earlier opsets always return the full shape. Added a test verifying opset 14 ignores start/end.
